custom/icds_reports/static/js/directives/location-filter/location-filter.directive.js

```diff
@@ -458,6 +458,17 @@ function LocationFilterController($rootScope, $scope, $location, $uibModal, loca
     vm.closeLocationChoices = function () {
         vm.showLocationChoices = false;
     };
+    $scope.$on('request_filter_data', function () {
```

Comment:
lot of message passing in this workflow, but i think it's still saner to follow than how it works on web.

the `mobile-filter.controller` requests data from the sub filters (just location for now). then compiles the results and republishes them together where they are picked up by other listeners (currently just `program-summary`).